build.gradle

```diff
@@ -71,3 +74,70 @@ spotless {
         eclipse().configFile rootProject.file('.eclipseformat.xml')
     }
 }
+
+task sourcesJar(type: Jar) {
+    classifier = 'sources'
+    from sourceSets.main.allJava
+}
+
+task javadocJar(type: Jar) {
+    classifier = 'javadoc'
+    from javadoc.destinationDir
+}
+
+publishing {
+    publications {
+        shadow(MavenPublication) {
+            project.shadow.component(it)
+            groupId = 'com.amazon.opendistroforelasticsearch'
+            artifactId = 'common-utils'
+
+            artifact sourcesJar
+            artifact javadocJar
+
+            pom {
+                name = "Open Distro for Elasticsearch Common Utils"
+                packaging = "jar"
+                url = "https://github.com/opendistro-for-elasticsearch/common-utils"
+                description = "Open Distro for Elasticsearch Common Utils"
+                scm {
+                    connection = "scm:git@github.com:opendistro-for-elasticsearch/common-utils.git"
+                    developerConnection = "scm:git@github.com:opendistro-for-elasticsearch/common-utils.git"
+                    url = "git@github.com:opendistro-for-elasticsearch/common-utils.git"
+                }
+                licenses {
+                    license {
+                        name = "The Apache License, Version 2.0"
+                        url = "http://www.apache.org/licenses/LICENSE-2.0.txt"
+                    }
+                }
+                developers {
+                    developer {
+                        id = "amazonwebservices"
+                        organization = "Amazon Web Services"
+                        organizationUrl = "https://aws.amazon.com"
+                    }
+                }
+            }
+        }
+    }
+
+    repositories {
+        maven {
+            name = "sonatype-staging"
+            url "https://aws.oss.sonatype.org/service/local/staging/deploy/maven2"
+            credentials {
+                username project.hasProperty('ossrhUsername') ? project.property('ossrhUsername') : ''
+                password project.hasProperty('ossrhPassword') ? project.property('ossrhPassword') : ''
+            }
+        }
+    }
+
+    gradle.startParameter.setShowStacktrace(ShowStacktrace.ALWAYS)
+    gradle.startParameter.setLogLevel(LogLevel.DEBUG)
```

Comment:
Why are these in the "publishing" block? They seem like they should be top-level directives.